improver/precipitation_type/freezing_rain.py

```diff
@@ -268,7 +269,7 @@ def _calculate_freezing_rain_probability(
 
         return freezing_rain
 
-    def process(self, input_cubes: CubeList) -> Cube:
+    def process(self, *input_cubes: Union[Cube, CubeList]) -> Cube:
```

Comment:
Technically this can parse a cube, but a cube cannot contain the three diagnostics required, so the plugin will fail. Should we be listing what can technically be handled, or what is actually needed in this typing statement?

Reply:
This is the type of each positional argument (note the asterisk).  That is, each positional argument can be either a cube or cubelist.  The documentation should then stipulate that we expect 3 cubes amongst whatever form of input is provided.
I'll flesh out the documentation here 👍

Reply:
actually the docs do already detail what cubes are expected 🤷

Reply:
Yeah, you and I clearly think differently, which is why I think you've not followed me here.
I'm saying that the only valid (as in, will actually make the code work) input to this process method is a cubelist. You can technically provide a cube, but you are guaranteed that this plugin will fail as it will not find the necessary inputs.

So my question was whether this typing information should cover all the things that can be passed in (cube, cubelist, str, donkey) or only what can be passed in that will lead to a successful output (cubelist). I'd hedge to the latter, but that's because I think of what the code is trying to do, rather than what it can technically handle, which is why I was asking for your view of it.

Reply:
Ah I see.  In that case, yes my view has been that the type should be what it will accept as input rather than being a reflection of what internal type coercion might be occurring internally.  Type hints also aid IDEs.

That being said, it is a balance between strictness and practicality.  It's often clearer to specify only the types we explicitly expect to handle (we needn't specify every possible type under the sun).